base/linalg/eigen.jl

```diff
@@ -67,8 +67,10 @@ function eig(A::Union{Number, AbstractMatrix}; kwargs...)
     F.values, F.vectors
 end
 #Calculates eigenvectors
-eigvecs(A::Union{Number, AbstractMatrix}, args...; kwargs...) = eigfact(A, args...; kwargs...)[:vectors]
+eigvecs(A::Union{Number, AbstractMatrix}, args...; kwargs...) = eigvecs(eigfact(A, args...; kwargs...))
+eigvecs{T,V,S,U}(F::Union(Eigen{T,V,S,U}, GeneralizedEigen{T,V,S,U}) = F[:vectors]::S
 
+eigvals{T,V,S,U}(F::Union(Eigen{T,V,S,U}, GeneralizedEigen{T,V,S,U}) = F[:values]::U
```

Comment:
Missing closing parenthesis here and I think we now prefer `Union{}` over `Union()`.